### PR TITLE
add more time before next EBD call

### PR DESCRIPTION
**Pull request by oliver-walker-hms**
data was still getting missed when start and end time are a second apart

### CODE REVIEW

**Comment by oliver-walker-hms** [comment]
I've been testing this and it was not working consistently with 1 second between the start and end time. It also seems to depend on how long the flexy has been running since I only saw this behavior after doing a reset on my flexy. Looks good with 2 seconds min.

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java [hunk 1]
```diff
@@ -238,19 +238,21 @@ public static synchronized ArrayList getFifoNextSpanData(
...
-  private static boolean sameSecond(long startTimeMillis, long endTimeMillis) {
-    boolean isSameSecond = false;
+  private static boolean enoughSeconds(long startTimeMillis, long endTimeMillis) {
+    boolean enoughTimePassed = true;
     long oneSecondMillis = 1000;
-    if ((startTimeMillis / oneSecondMillis) == (endTimeMillis / oneSecondMillis)) {
-      isSameSecond = true;
+    long secondsPassed = (endTimeMillis / oneSecondMillis) - (startTimeMillis / oneSecondMillis);
+
+    if (secondsPassed < 2) {
```
Comment (alexjhawk) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: 2 is a magic number

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java (hunk 1 above)
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: I think there should be a comment associated with the '2' number. It's explained a bit in this PR, but just looking at the code itself does not show why the 2 seconds value was chosen.

**Comment by alexjhawk** [comment]
The method _enoughSeconds_ seems like it should also have a warning or note explaining the reason that 'enough time' needs to pass before another EBD call, and what the repercussions are if that does not happe

**Comment by oliver-walker-hms** [comment]
I've added a large comment block in to the method explaining this which i'll push up shortly. You meant comments when you say a warning, right? The method probably should not log out any warnings about it since that would be very confusing for a user to see.

**Comment by TomKimsey** [comment]
Commit message and body dont follow HMS guidelines 
https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.wiki/tab::78561cc3-4c3b-4047-b463-167a3831b5ba?context=%7B%22subEntityId%22%3A%22%7B%5C%22pageId%5C%22%3A5%2C%5C%22sectionId%5C%22%3A16%2C%5C%22origin%5C%22%3A2%7D%22%2C%22channelId%22%3A%2219%3Accb41d2097924ef6aa065081838e1e83%40thread.skype%22%7D&tenantId=5c7c1590-4488-4e42-bc9c-15218f8ac994

**Comment by alexjhawk** [comment]
> I've added a large comment block in to the method explaining this which i'll push up shortly. You meant comments when you say a warning, right? The method probably should not log out any warnings about it since that would be very confusing for a user to see.

Sorry, I see how that could be confusing. Yes, I'm referring to a comment that acts as a disclaimer/warning

**Review by alexjhawk — DISMISSED** [review]

**Review by TomKimsey — COMMENTED** [review]
First line of commit title and body should be capitalized.

**Comment by oliver-walker-hms** [comment]
the commit message capitalization is fixed. I added that into the wiki as well under guidelines since I didn't see it on there

**Comment by TomKimsey** [comment]
> the commit message capitalization is fixed. I added that into the wiki as well under guidelines since I didn't see it on there

I added a commit msg hook Jon created to the wiki, that applies additional rules. There are some rules that are not explicit in the official hms guidelines but if you look at the template they have they are there. Capitalization of first word, no punctuation of title, punctuation in body.

**Review by alexjhawk — DISMISSED** [review]

**Review by TomKimsey — COMMENTED** [review]
just remove the period at the end of the title and you are good.

**Review by TomKimsey — APPROVED** [review]

**Review by alexjhawk — APPROVED** [review]

**Pull request merged**